### PR TITLE
YEP 64 API functions

### PR DESCRIPTION
**Pull request by alexis-moins**
- feat(composables): added useDatabase, useToken and useUtils composables, updated READMEs
- feat(middlewares): added authenticate middleware

### CODE REVIEW

File: back/src/composables/useDatabase.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  const account = await findAccountById(payload.id);
+  if (!account || account.email !== payload.email) return null;
+
+  return account;
+};
+
+const useDatabase = () => ({
+  database,
+
```
Comment (ASparton) [inline back/src/composables/useDatabase.ts:37]: Prettier did not automatically delete this empty line?

File: back/src/composables/useDatabase.ts
Comment (ASparton) [inline back/src/composables/useDatabase.ts:1]: Nice and simple functions that can be easily imported everywhere, like that

File: back/src/composables/useUtils.ts
Comment (ASparton) [inline back/src/composables/useUtils.ts:1]: I'm gonna copy that for the front :D

**Review by ASparton — APPROVED** [review]
Simple, straightforward. Like that

File: back/src/composables/useUtils.ts
Comment (alexis-moins) [reply back/src/composables/useUtils.ts:1]: Hell yeah 🤟🏻

File: back/src/composables/useDatabase.ts (hunk 1 above)
Comment (alexis-moins) [reply back/src/composables/useDatabase.ts:37]: Nope, it seems to have stayed to way I wrote it, even after prettier

File: back/src/middlewares/authenticate.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      success: false,
+      message: 'Unauthenticated.',
+    });
+
+  const account = await findAccountByToken(token);
+  if (!account)
+    return res.status(401).json({
+      success: false,
+      message: 'Unauthenticated.',
+    });
```
[the remark this answers is not in the document]
Comment (alexis-moins) [reply back/src/middlewares/authenticate.ts:26]: So you want a different message when the token is missing and when the token is invalid, is that it ?

File: back/README.md [hunk 3]
```diff
@@ -10,8 +10,9 @@ You will need [docker](https://www.docker.com) and [just](https://github.com/cas
 To bootstrap the application, install the dependencies using:
 `yarn install` or `npm install`
 
-You will also need a `.env` containing the `DATABASE_URL` variable.
-When working in the dev environment, the variable must be set to `mysql://root:root@database/gigz?connect_timeout=300`
+You will also need a `.env` containing the variables:
+- `DATABASE_URL` => default to `mysql://root:root@database/gigz?connect_timeout=300`
+- `TOKEN_SECRET`
```
Comment (Xerataw) [inline back/README.md:15]: Maybe you can rename it 'TOKEN_SALT'

Comment (alexis-moins) [reply back/README.md:15]: That's a good idea! Nice catch 🌝

File: back/src/middlewares/authenticate.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (filveith) [reply back/src/middlewares/authenticate.ts:26]: Yes, that would be nice, so in case we forget to add the token we instantly know that we are dumb :))

**Review by Xerataw — APPROVED** [review]

**Review by Xerataw — APPROVED** [review]

File: back/README.md (hunk 3 above)
Comment (alexis-moins) [reply back/README.md]: Done!

**Pull request merged**